### DISCUSSION TITLE
URL Link in search in search results box is beeing blocked only in the user defined layout

### DISCUSSION THREAD

**Discussion by kaddl1328** [microsoft-search/pnp-modern-search#3630] [Q&A]
Hello community!
very new here...
I have the issue, that the URL datatypes in the result box won´t open correctly. A warning appears which says the datatyp is not safe.
However if I use the layout "List" there is no problem with opening this URL-file.

Thank you guys!
Katharina

**Comment by wobba** [comment]
What is the URL scheme you have an issue with?

**Reply by kaddl1328** [reply]
Sorry wobba, I don´t know yet what you mean by your question...

**Reply by wobba** [reply]
When you mouse over the link, is it `https://...`, `mailto:...` etc. Some uri schemes are blocked by the browser like `file:`?

**Reply by kaddl1328** [reply]
Hi wobba, 
when I hover over, there seems to be no problem in the first place.  But when you eventually click, you get the error.
![url path](https://github.com/microsoft-search/pnp-modern-search/assets/157533568/486a1cb7-2083-4353-a740-3adb8b4aec2f)
![url error](https://github.com/microsoft-search/pnp-modern-search/assets/157533568/8a62d9a1-6343-46e4-bb19-f8b2110d8030)

This happens, when I use the customized layout - if I use the list for instance - it works.
Is there a bug in the user-defined layout or will I have to correct something (in the code?) to make it work?

Thank you very much for your patience!

**Reply by wobba** [reply]
@kaddl1328 so it's a .url file - in the list layout, is the URL the same? You need to provide information on what is different on the scenarios and how to reproduce this - if someone is to look into it and figure out what happens.

**Reply by kaddl1328** [reply]
Hi wobba, yes it is always the same url. It seems to me, that the layout "custom" reactions differently then all the other layouts to the url files... so my solution is now, to simple change the layout to "detaillist" - the URL is working here and I wont get an error... you can see that in my screenshot. I wish I knew how I could change the layout "custom" so there would not be an error, since I like the layout better, but I cant find the solution therefore.... 
Does it also make a difference when I add a command in the URL like "&action=view?web=1"? I will try that as well...
Thank you!

![url opens](https://github.com/microsoft-search/pnp-modern-search/assets/157533568/f5f547c3-6584-4db9-9e4b-f829f1609390)

**Reply by wobba** [reply]
Now I get your issue.. it was not clear it applied to .url / link files. We have special handling of them, and I created an issue to improve this across all templates. #3655

**Reply by kaddl1328** [reply]
Hi wobba - thank you! 
Do you also know how I can open the URL directly in the preview and I avoid to have to click the open button first?

**Reply by wobba** [reply]
I added a comment to include using the URL on preview as well.
